### PR TITLE
Feature branch/tet 613 refactor investment project

### PR DESCRIPTION
**Pull request by Richard-Pentecost**
## Description of change

Refactor project investment pages so that the layout is rendered before the resource is loaded. Create new layouts for the investment pages to use so that inline resources can be rendered for better user experience if there are errors.

## Test instructions
These steps use 'Super Awesome Project' as an example investment project but you can use any investment project to test the changes.

To carry out these steps, run the dev environment locally.

1. From the Data Hub homepage, navigate to the [Investments page](http://localhost:3000/investments/projects?page=1&sortby=created_on%3Adesc) 

2. Select an Investment Project to load the **Project details** page e.g. [Super Awesome Project](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/details)

3. You should see the page layout load before the page resources i.e. you should be able to see the page header and left hand navigation whilst the Investment information is loaded (see screen shots below for an example). Loading wheels should appear inline.

4. To test that the page displays error messages inline when an investment cannot be loaded, update the ProjectId to an unrecognisable string
e.g. 

> const ProjectTeam = () => {
> // const { projectId } = useParams()
> const projectId = 'asdf'

5. Repeat the previous steps for the following pages within the investment project:
- [Project team](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team)
- [Edit client relationship management](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/edit-client-relationship-management)
- [Edit team members](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/edit-team-members)
- [Tasks](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/tasks?sortby=-created_on)
- [Interactions](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/interactions)
- [Propositions](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/propositions?sortby=-created_on&page=1)
- [Evaluations](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/evaluation)
- [Edit history](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/edit-history)
- [Evidence](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/evidence)

### Before
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with no errors:
Screenshot 2024-01-31 at 10 08 04

### After
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with no errors:
Screenshot 2024-01-31 at 10 09 59

### Before
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with errors:
Screenshot 2024-01-31 at 10 52 40

### After
Appearance of [Project Team page](http://localhost:3000/investments/projects/6baef797-1b9f-417c-87ef-6e480d1bfdb2/team) with errors:
Screenshot 2024-01-31 at 10 44 06

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [x] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/client/components/InvestmentLocalHeader/index.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+          )}
+        </BreadcrumbsWrapper>
+        <FlashMessages flashMessages={flashMessages} />
+        <InvestmentResource id={projectId}>
+          {(project) => (
+            <>
+              {/* <WatchTextContent
+                onTextContentChange={(text) => {
+                  document.title = text
+                }}
```
Comment (marijnkampf) [inline src/client/components/InvestmentLocalHeader/index.jsx:71]: Is this comment needed?

File: src/client/components/InvestmentLocalHeader/index.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+              >
+                {pageTitle} - {project.name} - Projects - Investments - DBT Data
+                Hub
+              </WatchTextContent> */}
```
Comment (marijnkampf) [inline src/client/components/InvestmentLocalHeader/index.jsx:75]: See above

File: src/client/modules/Investments/Projects/Evidence/ProjectEvidence.jsx [hunk 3]
```diff
@@ -5,110 +5,119 @@ import { LEVEL_SIZE } from '@govuk-react/constants'
...
+          //   project={project}
+          //   breadcrumbs={[
+          //     {
+          //       link: urls.investments.projects.details(project.id),
+          //       text: project.name,
+          //     },
+          //     { text: 'Evidence' },
+          //   ]}
+          //   pageTitle="Evidence"
+          // >
```
Comment (marijnkampf) [inline src/client/modules/Investments/Projects/Evidence/ProjectEvidence.jsx:45]: Remove commented out code.

File: src/client/components/InvestmentLocalHeader/index.jsx (hunk 1 above)
Comment (swenban) [reply src/client/components/InvestmentLocalHeader/index.jsx]: WatchTextContent was relocated to the investment layout so I've now removed this commented out code

File: src/client/components/InvestmentLocalHeader/index.jsx (hunk 2 above)
Comment (swenban) [reply src/client/components/InvestmentLocalHeader/index.jsx]: This has now been removed

File: src/client/modules/Investments/Projects/Evidence/ProjectEvidence.jsx (hunk 3 above)
Comment (swenban) [reply src/client/modules/Investments/Projects/Evidence/ProjectEvidence.jsx]: Thank you, this has now been removed

**Review by marijnkampf — APPROVED** [review]

**Review by oliverjwroberts — CHANGES_REQUESTED** [review]
The changes you have made all look good to me and are working as expected on my end. In fact, I really like how the pages now load. 

However, I did note that when testing if the error messages load with an unknown project ID, clicking on the Admin tab caused an error and the app to crash.

After clicking on the `Evidence` tab, the error messages appear as expected:

image

After clicking on the `Admin` tab, the app crashed:

image

Console output:
```
error: uncaughtException: Cannot destructure property 'id' of 'res.locals.investment' as it is undefined.
TypeError: Cannot destructure property 'id' of 'res.locals.investment' as it is undefined.
[nodemon] app crashed - waiting for file changes before starting...
```

This can also be replicated on the staging environment with a 504 error, so most likely a pre-existing issue.

I'd be happy to discuss further, but thought we might want to investigate some more before publishing this change as it could potentially increase the chance of users accidentally experiencing the issue.

**Review by oliverjwroberts — APPROVED** [review]

**Pull request merged**